// @clr/angular/esm2020/modal/modal-stack.service.mjs
/*
 * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { isPlatformBrowser } from '@angular/common';
import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
import { Keys } from '../utils/enums/keys.enum';
import { normalizeKey } from '../utils/focus/key-focus/util';
import * as i0 from "@angular/core";
export class ModalStackService {
    constructor(platformId) {
        this.platformId = platformId;
        this.modalStack = [];
        this.keyUpEventListener = this.onKeyUp.bind(this);
    }
    trackModalOpen(openedModal) {
        if (this.modalStack.includes(openedModal) === false) {
            this.modalStack.unshift(openedModal);
        }
        if (isPlatformBrowser(this.platformId)) {
            document.body.addEventListener('keyup', this.keyUpEventListener);
        }
    }
    trackModalClose(closedModal) {
        const closedModalIndex = this.modalStack.indexOf(closedModal);
        if (closedModalIndex > -1) {
            this.modalStack.splice(closedModalIndex, 1);
        }
        if (this.modalStack.length === 0 && isPlatformBrowser(this.platformId)) {
            document.body.removeEventListener('keyup', this.keyUpEventListener);
        }
    }
    onKeyUp(event) {
        if (this.modalStack.length && normalizeKey(event.key) === Keys.Escape) {
            this.modalStack[0].close();
        }
    }
}
ModalStackService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.2", ngImport: i0, type: ModalStackService, deps: [{ token: PLATFORM_ID }], target: i0.ɵɵFactoryTarget.Injectable });
ModalStackService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.2.2", ngImport: i0, type: ModalStackService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.2", ngImport: i0, type: ModalStackService, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }]; } });
