// @clr/angular/esm2020/utils/popover/popover-host.directive.mjs
/*
 * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, ElementRef } from '@angular/core';
import { POPOVER_HOST_ANCHOR } from '../../popover/common/popover-host-anchor.token';
import { ClrPopoverEventsService } from './providers/popover-events.service';
import { ClrPopoverPositionService } from './providers/popover-position.service';
import { ClrPopoverToggleService } from './providers/popover-toggle.service';
import { ClrStopEscapePropagationDirective } from './stop-escape-propagation.directive';
import * as i0 from "@angular/core";
import * as i1 from "./stop-escape-propagation.directive";
export class PopoverHostDirective {
}
PopoverHostDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.2", ngImport: i0, type: PopoverHostDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
PopoverHostDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.2.2", type: PopoverHostDirective, isStandalone: true, providers: [
        ClrPopoverToggleService,
        ClrPopoverEventsService,
        ClrPopoverPositionService,
        { provide: POPOVER_HOST_ANCHOR, useExisting: ElementRef },
    ], hostDirectives: [{ directive: i1.ClrStopEscapePropagationDirective }], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.2", ngImport: i0, type: PopoverHostDirective, decorators: [{
            type: Directive,
            args: [{
                    standalone: true,
                    providers: [
                        ClrPopoverToggleService,
                        ClrPopoverEventsService,
                        ClrPopoverPositionService,
                        { provide: POPOVER_HOST_ANCHOR, useExisting: ElementRef },
                    ],
                    hostDirectives: [ClrStopEscapePropagationDirective],
                }]
        }] });
